### PR TITLE
Expose collection metadata API documentation with Swagger

### PR DESCRIPTION
**Pull request by Ndpnt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,6 +93,7 @@
     "sib-api-v3-sdk": "^8.2.1",
     "simple-git": "^3.8.0",
     "swagger-jsdoc": "^6.2.8",
+    "swagger-ui-express": "4.6.2",
```
Comment (MattiSG) [inline package.json:96]: Why is this version locked? 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
...
 All changes that impact users of this module are documented in this file, in the [Common Changelog](https://common-changelog.org) format with some additional specifications defined in the CONTRIBUTING file. This codebase adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## Unreleased [minor]
+
+_Full changeset and discussions: [#1015](https://github.com/OpenTermsArchive/engine/pull/1015)._
+
+### Added
+
+- Expose collection metadata API documentation with a user-friendly interface
```
Comment (MattiSG) [inline CHANGELOG.md:11]: Let's avoid judgement (even positive) in our changelog.

```suggestion
- Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API
```

File: package.json (hunk 1 above)
Comment (Ndpnt) [reply package.json:96]: Because the version 4.6.3 is broken

**Review by MattiSG — COMMENTED** [review]
I get a blank page if I navigate to `http://localhost:3000/api/v1/docs/` after having run `npm install` and `npm start:api`.

**Comment by MattiSG** [comment]
(also, the server does not echo the URL, which would be useful, but that's out of scope I guess)

**Comment by Ndpnt** [comment]
> I get a blank page if I navigate to `http://localhost:3000/api/v1/docs/` after having run `npm install` and `npm start:api`.

Hum weird 🤔. I have the same error as you with Safari but it works with Chrome and Firefox.

**Comment by Ndpnt** [comment]
Ok, I found why. It's related to the way browsers handle `upgrade-insecure-requests directive`. 
See https://github.com/helmetjs/helmet/issues/429#issuecomment-1599789358:
> Chrome and Firefox don't think that localhost needs "upgrading" from HTTP to HTTPS, but Safari does.

**Comment by Ndpnt** [comment]
I fix the issue on Safari and I also change the Swagger UI url to `/` instead of `/docs` to load the Swagger UI docs by default.

**Comment by Ndpnt** [comment]
I tried the deployment on the experimentation server, you can check the Swagger UI here: http://54.37.5.206/api/v1/

File: src/api/routes/docs.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      });
+
+      describe('body response defines', () => {
+        let subject;
+
+        before(async () => {
+          subject = response.body;
+        });
+
+        it('openapi version', () => {
```
Comment (MattiSG) [inline src/api/routes/docs.test.js:33]:
```suggestion
        it('OpenAPI version', () => {
```

File: src/api/routes/index.js [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+      directives: defaultDirectives,
+    },
+  })); // Do not enable `upgrade-insecure-requests` directive set by Helmet for docs routes to ensure insecure requests won't be upgraded to secure requests for swaggerUI assets; see https://github.com/center-for-threat-informed-defense/attack-workbench-rest-api/issues/96#issuecomment-924193910 and https://github.com/scottie1984/swagger-ui-express/issues/212#issuecomment-825803088
+
+  router.use(docsRouter(basePath));
+
+  router.use(helmet()); // then, enable all `helmet` HTTP response headers for all others routes
+
+  router.get('/', (req, res) => {
+    res.json({ message: 'Welcome to an instance of the Open Terms Archive API. Documentation is available at /docs' });
```
Comment (MattiSG) [inline src/api/routes/index.js:26]:
```suggestion
    res.json({ message: 'Welcome to an instance of the Open Terms Archive API. Documentation is available at /docs. Learn more on Open Terms Archive on https://opentermsarchive.org.' });
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,7 +2,13 @@
...
 All changes that impact users of this module are documented in this file, in the [Common Changelog](https://common-changelog.org) format with some additional specifications defined in the CONTRIBUTING file. This codebase adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## Unreleased
+## Unreleased [minor]
+
+_Full changeset and discussions: [#1015](https://github.com/OpenTermsArchive/engine/pull/1015)._
+
+### Added
+
+- Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API
```
Comment (MattiSG) [inline CHANGELOG.md:11]: I believe we are missing some breaking changes announcements, don't we? 🙂

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,10 @@ _Full changeset and discussions: [#1015](https://github.com/OpenTermsArchive/eng
 
 - Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API
 
+### Changed
+
+- Change path from `/services/:serviceId` to `/service/:serviceId` for direct access to a `service` resource; update paths accordingly in your own codebase
```
Comment (MattiSG) [inline CHANGELOG.md:15]: Always put breaking changes on top (before additions).
```suggestion
- **Breaking** Change path from `/services/:serviceId` to `/service/:serviceId` for direct access to a `service` resource; update paths accordingly in your own codebase
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,6 +10,10 @@ _Full changeset and discussions: [#1015](https://github.com/OpenTermsArchive/eng
 
 - Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API
```
Comment (MattiSG) [inline CHANGELOG.md:11]: > Was the OpenAPI documentation already there before this changeset? 🤔 Didn't it move from `specs` to `docs`?

**Review by MattiSG — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,7 +2,18 @@
...
+_Full changeset and discussions: [#1015](https://github.com/OpenTermsArchive/engine/pull/1015)._
+
+### Changed
+
+- **Breaking:** Change path from `/services/:serviceId` to `/service/:serviceId` for direct access to a `service` resource; update paths accordingly in your own codebase
+- **Breaking:** Change path from `/specs` to `/docs` for OpenAPI specification; update paths accordingly in your own codebase and set the HTTP header `Accept: application/json`
+
+### Added
+
+- Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API
```
Comment (MattiSG) [inline CHANGELOG.md:16]:
```suggestion
- Embed [Swagger UI](https://swagger.io) for graphical user interface documentation of the API; access it on `/docs`
```

**Review by MattiSG — APPROVED** [review]
🎉 Perfect!!

**Pull request merged**